src/core.ts

```diff
@@ -1138,9 +1138,33 @@ function applyHeadersMut(targetHeaders: Headers, newHeaders: Headers): void {
   }
 }
 
+const SENSITIVE_HEADERS = new Set(['authorization', 'api-key']);
+
 export function debug(action: string, ...args: any[]) {
   if (typeof process !== 'undefined' && process?.env?.['DEBUG'] === 'true') {
-    console.log(`OpenAI:DEBUG:${action}`, ...args);
+    const modifiedArgs = args.map((arg) => {
+      if (!arg) {
+        return arg;
+      }
+
+      const modifiedArg = { ...arg };
+      // Check for sensitive headers in request body 'headers' object
+      if (modifiedArg['headers']) {
+        for (const header in modifiedArg['headers']) {
+          if (SENSITIVE_HEADERS.has(header.toLowerCase())) {
+            modifiedArg['headers'][header] = 'REDACTED';
```

Comment:
I think this will still mutate the original `headers` property as `...` is just a shallow clone

Reply:
That's a good point, thank you!